distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java

```diff
@@ -151,5 +183,122 @@ private Response respond(Response response) {
 
             return response;
         }
+
+        /**
+         * Holds byte offset data specified by the request's range header and any related parsing
+         * methods.
+         */
+        private class ByteRange {
+            public int start;
+
+            public int end;
+
+            public ByteRange(int start, int end) {
+                this.start = start;
+                this.end = end;
+            }
+
+            /**
+             * Tokenizes the starting and ending bytes values from a range header. The stub server only
+             * advertising supporting byte offsets specifically, so this function assumes that byte
+             * ranges are provided and does not evaluate the unit of measurement
+             *
+             * @param rangeHeaderValue          Value of the range header sent in the request
+             * @param totalSizeOfProductInBytes Total size of the message to be returned. This is
+             *                                  not the partial content size, but the FULL size of
+             *                                  the product.
+             */
+            public ByteRange(String rangeHeaderValue, int totalSizeOfProductInBytes) {
+                // extract bytes
+                String startToken = StringUtils.substringBetween(rangeHeaderValue, "=", "-");
+                String endToken = StringUtils.substringAfter(rangeHeaderValue, "-");
+
+                // range offsets can be blank to indicate "until beginning" or "until end" of data.
+                if (StringUtils.isBlank(startToken)) {
+                    start = 0;
+                } else {
+                    start = Integer.parseInt(startToken);
+                }
+                if (StringUtils.isBlank(endToken)) {
+                    end = totalSizeOfProductInBytes - 1;
+                } else {
+                    end = Integer.parseInt(endToken);
+                }
+            }
+
+            public String contentRangeValue() {
+                return "bytes " + start + "-" + end + "/" + responseMessage.length;
```

Comment:
We could add a way to set `/*` for testing purposes, but I don't think it's necessary. The old range-header support logic in DDF only looked to make sure that the `Accept-Ranges` header was present and assumed that a `Partial Content` response would return the right range. While we should probably update this to at least verify the range returned is correct, the I don't think the `/(content-size | *)` is vital enough to warrant extra testing since we probably won't even be using it in our retrieval logic.